server/sonar-server/src/main/java/org/sonar/server/platform/web/MasterServletFilter.java

```diff
@@ -134,4 +137,25 @@ public void addFilter(Filter filter) {
       filters.add(filter);
     }
   }
+
+  private static class ServletRequestWrapper extends HttpServletRequestWrapper {
+
+    ServletRequestWrapper(HttpServletRequest request) {
+      super(request);
+    }
+
+    @Override
+    public HttpSession getSession(boolean create) {
+      throw notSupported();
+    }
+
+    @Override
+    public HttpSession getSession() {
+      throw notSupported();
+    }
+
+    private static UnsupportedOperationException notSupported() {
+      return new UnsupportedOperationException("Sessions are disabled in order to support clustering of web servers without stateful server sessions");
```

Comment:
I suggest to not mention clustering: "Sessions are disabled so that web server is stateless".